File: src/main/kotlin/inflearn/java/ch01_string/_1_7_회문_문자열.java
package inflearn.java.ch01_string;

import java.util.Scanner;

public class _1_7_회문_문자열 {
    public static String solution(String str) {
        StringBuilder sb = new StringBuilder(str);
        sb.reverse();

        if(sb.toString().equalsIgnoreCase(str))
            return "YES";
        else
            return  "NO";
    }

    public static String solution2(String str) {
        int len = str.length();
        str = str.toUpperCase();
        for(int i = 0; i < len/2; i++) {
            if(str.charAt(i) != str.charAt(len - i - 1))
                return "NO";
        }
        return "YES";
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String input = sc.next();
        System.out.println(solution(input));
    }
}

File: src/main/kotlin/inflearn/1_7 회문 문자열.kt
package inflearn

import java.util.*

object `1_7 회문 문자열` {
    fun solution(str: String?): String? {
        val sb = StringBuilder(str)
        sb.reverse()
        return if (sb.toString().equals(str, ignoreCase = true)) "YES" else "NO"
    }

    fun solution2(str: String): String? {
        var str = str
        val len = str.length
        str = str.uppercase(Locale.getDefault())
        for (i in 0 until len / 2) {
            if (str[i] != str[len - i - 1]) return "NO"
        }
        return "YES"
    }

    @JvmStatic
    fun main(args: Array<String>) {
        val sc = Scanner(System.`in`)
        val input = sc.next()
        println(solution(input))
    }
}